alphagov/paas-cf
# [#114110115] Use bosh interpolate to generate secrets

What
----

We want to move as many secrets as possible to be generated with
`bosh interpolate` as it is done in the upstream cf-deployment,
keeping the upstream names.

See the commits for details about the change.


How to review
-------------

 - Code review
 - Deploy from master
 - Deploy from this branch: No changes should be done in the
   deployment or in terraform. Only a UAA client removed: tcp_router
 - Test secret rotation (See readme)

    Only the mechanism of rotation has changed. We shouldn't have changed what gets rotated. The below points are deliberately not prescriptive about how you should test this.
    - check we rotate everything we used to rotate
    - check we haven't started rotating things we never used to rotate.

 - Test delete and create (can be auto-delete)


Who can review
--------------

Anyone but @henrytk  or @keymon
